### DISCUSSION TITLE
[EditorBrowsable(EditorBrowsableState.Dot)]

### DISCUSSION THREAD

**Discussion by kefyru** [dotnet/roslyn#60858] [Ideas]
I'm working with project which generates thousands of types (it is big database).
For example there is essence 'Product' with classes:
```cs
public static partial class Ref {
   public class Product { ... }
   public class Invoice { 
       public class Products { ... }
   }
   ...
}

public static partial class Obj {
   public class Product { ... }
   ...
}

public static partial class Meta {
   public class Product { ... }
   ...
}

public static partial class DB {
   public class Product { ... }
   ...
}
```

Typing 'Ref.Product' IDE suggest autocompletion:
```
Ref. Product
Obj. Product
Meta.Product
DB. Product
Ref.Invoice.Products
Obj.Invoice.Products
Meta.Invoice.Products
... e.t.c.
``` 

I tried to use `[EditorBrowsable(EditorBrowsableState.Never)]` in various places. It hides classes. But not as i want.
Is it possible to add `EditorBrowsableState.Dot` state.
Behavior:
- typing `product` will not suggest anything
- typing `Ref.product` will suggest only matching types direct from `Ref`, not `Ref.*.*`

**Comment by CyrusNajmabadi** [comment]
Can you show a picture of what you're seeing today?

**Reply by kefyru** [reply]
<img width="633" alt="Снимок экрана 2022-04-22 в 10 35 35" src="https://user-images.githubusercontent.com/29143718/164629207-5a38f6c4-f11b-49b9-910c-1e1ffe1549af.png">

**Reply by kefyru** [reply]
<img width="507" alt="Снимок экрана 2022-04-22 в 10 38 59" src="https://user-images.githubusercontent.com/29143718/164632078-00655cea-d4ae-436f-8b49-c787e403024d.png">

**Comment by kefyru** [comment]
I want to show in root context just type from `Db`, and other types (from Obj., Ref., Meta., e.t.c) i want to show only after typing dot.
I have thousands of types totally and IDE takes some time to find all of Subtypes. I think if big part of them will be hidden (with `[EditorBrowsable(EditorBrowsableState.Dot)]`) IDE will be easier to find matching types. And we will be able to customise type structure more convenient.

**Reply by kefyru** [reply]
usage:
`[EditorBrowsable(EditorBrowsableState.Dot)]`
or
`[EditorBrowsable(EditorBrowsableState.Dot | EditorBrowsableState.Advanced)]`

**Reply by CyrusNajmabadi** [reply]
What tool are you using here?  This doesn't look like Roslyn AFAICT.

**Reply by kefyru** [reply]
Rider

**Reply by CyrusNajmabadi** [reply]
Rider is not powered by Roslyn.  You'llneed to bring this up with them.
